### PR TITLE
apfree-wifidog: add new package

### PR DESCRIPTION
**Pull request by liudf0716**
Signed-off-by: Dengfeng Liu liudengfeng@kunteng.org.cn

Maintainer: me / @liudf0716
Compile tested: x86, x86_64, CentOS 7,  OpenWrt master 7e73e912
Run tested:  arm, R7800, OpenWrt master 7e73e912

Description:
comparing with wifidog, apfree-wifidog is an impoved solution,  which is more stable and more efficient.  apfree-wifidog was widely used at business scene especially in china.

### CODE REVIEW

**Comment by karlp** [comment]
At the very least, these should all be squished to a single clean commit

File: net/apfree-wifidog/Makefile [hunk 1]
```diff
@@ -0,0 +1,72 @@
+#
+# Copyright (C) 2006,2015 OpenWrt.org
```
Comment (karlp) [inline net/apfree-wifidog/Makefile]: No, either drop this, or use your own. (unless you actually have copyright assignment on file with openwrt)

File: net/apfree-wifidog/Makefile [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+include $(INCLUDE_DIR)/package.mk
+include $(INCLUDE_DIR)/cmake.mk
+
+define Package/$(PKG_NAME)
+  SUBMENU:=Captive Portals
+  SECTION:=net
+  CATEGORY:=Network
+  DEPENDS:=+zlib +iptables-mod-extra +iptables-mod-ipopt +kmod-ipt-nat +iptables-mod-nat-extra \
+           +libpthread +libopenssl +@OPENSSL_WITH_EC +@OPENSSL_WITH_DEPRECATED +@OPENSSL_WITH_PSK +libjson-c +ipset +libip4tc +libevent2 +libevent2-openssl \
+		   +fping +libmosquitto +libuci
```
Comment (karlp) [inline net/apfree-wifidog/Makefile]: clean up your line wrapping here.

File: net/apfree-wifidog/Makefile [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+	of their Internet connection.
+    It's enhanced wifidog
+endef
+
+define Package/$(PKG_NAME)/install
+	$(INSTALL_DIR) $(1)/usr/bin
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/wifidog $(1)/usr/bin/
+	$(INSTALL_BIN) $(PKG_INSTALL_DIR)/usr/bin/wdctl $(1)/usr/bin/
+	$(INSTALL_DIR) $(1)/usr/lib
+	$(CP) $(PKG_INSTALL_DIR)/usr/lib/libhttpd.so* $(1)/usr/lib/
```
Comment (karlp) [inline net/apfree-wifidog/Makefile]: what library is this? isn't there an existing libhttpd?

**Comment by karlp** [comment]
The inclusion of specific certs and keys looks dubious, and it should probably not default to connecting to your backend.

File: net/apfree-wifidog/files/wifidog.init [hunk 4]
```diff
@@ -0,0 +1,248 @@
+#!/bin/sh /etc/rc.common
+# Copyright (C) 2006 OpenWrt.org
```
Comment (karlp) [inline net/apfree-wifidog/files/wifidog.init]: no. either drop or replace with your own.

File: net/apfree-wifidog/files/wifidog.init [hunk 5]
```diff
@@ -0,0 +1,248 @@
...
+	local set_no_auth=`[ -n "$no_auth"  ] && echo "NoAuth"  $no_auth`
+	local set_firewall_rule_global=`config_list_foreach "$cfg" "firewall_rule_global" echo_firewall_rule`
+	local set_firewall_rule_validating_users=`config_list_foreach "$cfg" "firewall_rule_validating_users" echo_firewall_rule`
+	local set_firewall_rule_known_users=`config_list_foreach "$cfg" "firewall_rule_known_users" echo_firewall_rule`
+	local set_firewall_rule_auth_is_down=`config_list_foreach "$cfg" "firewall_rule_auth_is_down" echo_firewall_rule`
+	local set_firewall_rule_unknown_users=`config_list_foreach "$cfg" "firewall_rule_unknown_users" echo_firewall_rule`
+	local set_firewall_rule_locked_users=`config_list_foreach "$cfg" "firewall_rule_locked_users" echo_firewall_rule`
+	local set_apple_cna=`[ -n "$apple_cna"  ] && echo "BypassAppleCNA" $apple_cna`
+	local set_update_domain_interval=`[ -n "$update_domain_interval" ] && echo "UpdateDomainInterval" $update_domain_interval`
+	local set_dns_timeout=`[ -n "$dns_timeout" ] && echo "DNSTimeout" $dns_timeout`
```
Comment (karlp) [inline net/apfree-wifidog/files/wifidog.init]: this is extremely complicated, are you sure it needs to be like this? you're invoking dozens of extra shells.

File: net/apfree-wifidog/Makefile (hunk 3 above)
Comment (liudf0716) [reply net/apfree-wifidog/Makefile]: libhttpd was included in apfree-wifidog project, it's not out-side library depends

File: net/apfree-wifidog/files/wifidog.init (hunk 5 above)
Comment (liudf0716) [reply net/apfree-wifidog/files/wifidog.init]: all these script was used to make user utilizing apfree-wifidog more easy

File: net/apfree-wifidog/Makefile (hunk 1 above)
Comment (liudf0716) [reply net/apfree-wifidog/Makefile]: ok, I'll modify these copyright information

**Comment by liudf0716** [comment]
@karlp sorry I must close this pr, cause my stupid mistake, i forgot checkouting a  new branch after forking packages

**Pull request closed without merging**